agent/agent.go

```diff
@@ -462,23 +464,27 @@ func (a *Agent) listenAndServeDNS() error {
 			defer a.wgServers.Done()
 			err := s.ListenAndServe(addr.Network(), addr.String(), func() { notif <- addr })
 			if err != nil && !strings.Contains(err.Error(), "accept") {
+				errCh <- err
 				a.logger.Printf("[ERR] agent: Error starting DNS server %s (%s): %v", addr.String(), addr.Network(), err)
```

Comment:
I wonder why this log message wasn't sufficient before - it seems it's already printing the same info that the returned err?

Maybe it was due to a race or something?

Should we consider _not_ logging here at all now - if this is ever seen it seems confusing that the same error will be output in multiple places?

Reply:
Its due to how we set up the logging. We have a Gated Logger which just buffers logs until the first flush is called on it. Then it streams them regularly.

Our agent once everything is setup and the general parameters are output issues the flush to cause outputting of the logs. That is when things go okay. Before it was timing out and never calling flush so those messages never got output.

One other potential solution was to call flush in the error case as well but I thought it was better to more clearly explain the problems rather than have users dig through those logs.

I agree though that the error log isn't needed anymore.